va/validation-authority.go

```diff
@@ -169,9 +223,20 @@ func (va ValidationAuthorityImpl) validateSimpleHTTP(identifier core.AcmeIdentif
 		// We don't expect to make multiple requests to a client, so close
 		// connection immediately.
 		DisableKeepAlives: true,
+		// Intercept Dial in order to connect to the IP address we
+		// select.
+		Dial: dialer,
 	}
+
 	logRedirect := func(req *http.Request, via []*http.Request) error {
```

Comment:
logRedirect is responsible for enforcing the max number of redirects, right? Can you add an error when challenge.Redirects is greater than ten elements, and a corresponding test function?